xla/stream_executor/gpu/read_numa_node.h

```diff
@@ -16,13 +16,16 @@ limitations under the License.
 #ifndef XLA_STREAM_EXECUTOR_GPU_READ_NUMA_NODE_H_
 #define XLA_STREAM_EXECUTOR_GPU_READ_NUMA_NODE_H_
 
+#include <optional>
 #include <string>
 
 namespace stream_executor::gpu {
 
 // Attempts to read the NUMA node corresponding to the GPU device's PCI bus out
-// of SysFS. Returns -1 if it cannot.
-int ReadNumaNode(const std::string& pci_bus_id, int device_ordinal);
+// of SysFS. Returns an empty optional if no value could be determined, returns
+// tsl::port::kNUMANoAffinity if the kernel reports a negative value.
+std::optional<int> ReadNumaNode(const std::string& pci_bus_id,
```

Comment:
Nit: Since you're already touching that function, can you make `pci_bus_id` a `std::string_view`?